src/Service.Host/client/src/components/Navbar.js

```diff
@@ -1,58 +1,163 @@
 import React, { useState, Fragment } from 'react';
 import PropTypes from 'prop-types';
 import { withStyles } from '@material-ui/core/styles';
-import AppBar from '@material-ui/core/AppBar';
-import Tabs from '@material-ui/core/Tabs';
-import Tab from '@material-ui/core/Tab';
+import {
+    Tabs,
+    Tab,
+    AppBar,
+    ClickAwayListener,
+    Menu,
+    MenuItem,
+    Toolbar,
+    Grid,
+    Typography
+} from '@material-ui/core';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import Panel from './Panel';
 
-const styles = {
+const styles = theme => ({
     root: {
-        width: '100%',
         position: 'absolute',
-        top: 0
-    },
-    tab: {
-        minWidth: 20
+        width: '100%',
+        top: 0,
+        zIndex: 10
     },
     tabLabel: {
         fontSize: '12px'
+    },
+    panel: {
+        position: 'relative'
+    },
+    menuButton: {
+        marginLeft: -12,
+        marginRight: 20
+    },
+    fullHeight: {
+        ...theme.mixins.toolbar,
+        minWidth: '100px'
+    },
+    toolbar: {
+        paddingLeft: 0,
+        paddingRight: 0
+    },
+    tabs: {
+        ...theme.mixins.toolbar
+    },
+    container: {
+        width: '100%'
     }
-};
+});
 
-function Navbar({ classes, menu, loading, history }) {
+function Navbar({ classes, menu, loading, username }) {
     const [selected, setSelected] = useState(false);
-
-    const handleChange = (event, value) => {
+    const [anchorEl, setAnchorEl] = useState();
+    if (menu) {
```

Comment:
if menu is an array then it will always be true... sorry, that could be clearer - only if menu is undefined will it not enter.. is that what you intended?

Reply:
Menu is an object (I think) which can be null if the request fails

Reply:
Oh no wait, menu is an array